packages/salesforcedx-vscode-apex/src/commands/metadataOrchestrator.ts

```diff
@@ -42,182 +44,26 @@ export interface Parameter {
-
-      // Stop once the closing parenthesis is reached
-      if (line.includes(') {')) {
-        break;
-      }
-    }
-
-    const methodMetadata = this.parseMethodSignature(methodSignature, isAuraEnabled);
-    if (!methodMetadata.isAuraEnabled) {
-      throw new Error(nls.localize('not_aura_enabled', methodMetadata.name));
-    }
-    return methodMetadata;
-  };
-
-  public extractAllMethodsMetadata = async (
-    sourceUri: vscode.Uri | undefined
-  ): Promise<MethodMetadata[] | undefined> => {
-    let lines;
-    let className;
-    if (sourceUri) {
-      const path = sourceUri?.path.toString();
-      className = path!
-        .substring(path!.lastIndexOf(process.platform === 'win32' ? '\\' : '/') + 1)
-        .split('.')
-        .shift();
-      const fileContent = await vscode.workspace.fs.readFile(sourceUri!);
-      const fileText = Buffer.from(fileContent).toString('utf-8');
-      lines = fileText.split('\n');
-    } else {
-      const editor = vscode.window.activeTextEditor;
-      if (!editor || !editor.document.fileName.endsWith('.cls')) {
-        notificationService.showErrorMessage(nls.localize('invalid_active_text_editor'));
-        return;
-      }
-
-      const document = editor.document;
-      const filePath = document.fileName;
-      className = filePath
-        .substring(filePath.lastIndexOf(process.platform === 'win32' ? '\\' : '/') + 1)
-        .split('.')
-        .shift();
-      lines = document.getText().split('\n');
+  public extractMetadata = async (
+    sourceUri: vscode.Uri | vscode.Uri[],
+    isMethodSelected: boolean = false
+  ): Promise<ApexClassOASEligibleResponse | undefined> => {
+    const isEligibleResponses = await this.validateEligibility(sourceUri, isMethodSelected);
+    if (!isEligibleResponses || isEligibleResponses.length === 0) {
+      throw new Error(nls.localize('validation_failed'));
     }
-    const metadataList: MethodMetadata[] = [];
-    let currentMethodSignature = '';
-    let isAuraEnabled = false;
-
-    for (let line of lines) {
-      line = line.trim();
-
-      // Detect @AuraEnabled annotation
-      if (line.includes('@AuraEnabled')) {
-        isAuraEnabled = true;
-      }
-
-      // Build the method signature
-      currentMethodSignature += ` ${line}`;
-      if (line.includes(') {') && currentMethodSignature.includes('(')) {
-        // Method signature is complete
-        if (isAuraEnabled) {
-          const methodMetadata = this.parseMethodSignature(currentMethodSignature, isAuraEnabled, className);
-          if (methodMetadata) {
-            metadataList.push(methodMetadata);
-          }
-          isAuraEnabled = false;
-        }
-
-        // Reset for the next method
-        currentMethodSignature = '';
+    if (!isEligibleResponses[0].isEligible) {
+      if (isMethodSelected) {
+        const name = isEligibleResponses?.[0]?.symbols?.[0]?.docSymbol.name;
+        throw new Error(nls.localize('not_aura_enabled', name));
```

Comment:
could the error be due to wrong access modifiers or so?

Reply:
I don't know, I think there's no way to tell with only `.isEligible`. We could modify the message to include that it's wither not auraenabled or private, or something like that

Reply:
Yeah we could provide a more general error message with help of Sonal.

Reply:
In addition to the selected method not being eligible, there is a chance the user did not select a method, chose gen for selected method, when they we actually trying to run for the entire class.

@CristiCanizales would you mind writing a spike work item to look into reporting of results from the eligibility call?